crates/noirc_evaluator/src/ssa/function.rs

```diff
@@ -56,15 +73,17 @@ impl SSAFunction {
         igen: &mut IRGenerator,
         env: &mut Environment,
     ) -> NodeId {
+        let otype = igen.context.functions[&func].result[0];
+        let ins_arguments = igen.expression_list_to_objects(env, arguments);
         let call_id = igen.context.new_instruction(
             NodeId::dummy(),
             NodeId::dummy(),
             node::Operation::Call(func),
-            node::ObjectType::NotAnObject, //TODO how to get the function return type?
+            otype,
```

Comment:
Are there instructions that use the `res_type` field from a function call? Without a results instruction it should have no result type of itself.

Reply:
Since we do not support multiple return values here, it makes sense to have the type of the return value in the res_type.
However not using it will probably make it easier to implement the multiple return values so may be I should put back the not_an_object.

Reply:
Finally I keep the return type, it is used I think for intrinsic calls. I do not want to break this right now, this will be  reviewed when we implement multiple return values.